hw/ip/i2c/doc/_index.md

```diff
@@ -1,3 +1,223 @@
+
+The I2C reads each format indicator from the head of FMT_FIFO, and processes them in turn.
+If none of the flags are set for the format format indicator none, the I2C FSM simply transmits the Format Byte onto the SCL and SDA pins according to the specification, waits for acknowledgement, and then proceeds to the next format indicator.
+The format flags modulate the behavior as follows.
+- READ (Corresponds to {{< regref FDATA.READ >}}):
+Signifies the Format Byte ({{< regref FDATA.FBYTE >}}) should be treated as an unsigned number, R, and prompts the state machine to read R bytes from the target device.
+Bytes read from the bus, are inserted into the RX FIFO where they can be accessed by software.
+A value of 0 is treated as a read of 256B.
+To read a larger byte stream, multiple 256B reads can be chained together using the RCONT flag.
+- RCONT (Corresponds to FIFO inputs {{< regref FDATA.RCONT >}}, Only used with READ):
+If RCONT is set, the Format Byte represents part of a longer sequence of reads, allowing for reads to be chained indefinitely.
+The RCONT flag indicates the the final byte returned with the current read
+should be responded to which an ACK, allowing the target to continue sending data.
+(Note that the first R-1 bytes read will still be acknowledged regardless
+of whether RCONT is asserted or not).
+- START (Corresponds to {{< regref FDATA.START >}}, Ignored when used with READ):
+Issue a START condition before transmitting the Format Byte on the bus.
+This flag may also be used to issue a repeated start condition.
+- STOP (Corresponds to {{< regref FDATA.STOP >}}):
+Issue a STOP signal after processing this current entry in the FMT FIFO.
+Note that this flag is not compatible with (READ & RCONT), and will cause bus conflicts.
+- NAKOK (Corresponds to {{< regref FDATA.NAKOK >}}, Not compatible with READ):
+Typically every byte transmitted must also receive an ACK signal, and the IP will raise an exception if no ACK is received.
+However, there are some I2C commands which do not require an ACK.
+In those cases this flag should be asserted with FBYTE indicating no ACK is expected and no interrupt should be raised if the ACK is not received. 
+
+### Timing Control Registers
+
+For Standard-mode, Fast-mode and Fast-mode Plus, the timing requirements for each transaction are detailed in Table 10 of the I2C specification.
+In order to claim complete compatibility at each mode, the state machine timings need to be adapted to whether there are Standard-mode, Fast-mode and Fast-mode Plus devices on the bus.
+Furthermore, depending on the actual capacitance of the bus, even a bus with all Fast-mode Plus capable devices may have to operate at slower speeds than 1MBaud.
```

Comment:
`1 Mbaud` (lower-case `b` for consistency with other uses in this document; the "official" symbol would be `1 MBd`, but I've not seen that used very widely)